src/sql-parser/src/parser.rs

```diff
@@ -5265,6 +5266,13 @@ impl<'a> Parser<'a> {
                 ObjectType::Table => ShowObjectType::Table,
                 ObjectType::View => ShowObjectType::View,
                 ObjectType::Source => ShowObjectType::Source,
+                ObjectType::Subsource => {
+                    self.expect_keyword(ON)?;
+
+                    ShowObjectType::Subsource {
+                        on_source: self.parse_raw_name()?,
+                    }
+                }
```

Comment:
Can you add some parser tests here: https://github.com/MaterializeInc/materialize/blob/main/src/sql-parser/tests/testdata/show

Reply:
Added